packages/react-components/src/components/FileDownloadCards.tsx

```diff
@@ -202,23 +212,30 @@ export const _FileDownloadCards = (props: _FileDownloadCards): JSX.Element => {
     <div style={fileDownloadCardsStyle} data-ui-id="file-download-card-group">
       <_FileCardGroup>
         {fileMetadata &&
-          fileMetadata.map((file) => (
-            <_FileCard
-              fileName={file.name}
-              key={file.name}
-              fileExtension={file.extension}
-              actionIcon={
-                showSpinner ? (
-                  <Spinner size={SpinnerSize.medium} aria-live={'polite'} role={'status'} />
-                ) : (
-                  <IconButton className={iconButtonClassName} ariaLabel={downloadFileButtonString()}>
-                    <DownloadIconTrampoline />
-                  </IconButton>
-                )
-              }
-              actionHandler={() => fileDownloadHandler(userId, file)}
-            />
-          ))}
+          fileMetadata
+            .filter((attachment) => {
+              /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+              return attachment.attachmentType === 'fileSharing';
+              return true;
+            })
+            .map((file) => (
+              <_FileCard
+                fileName={file.name}
+                key={file.name}
+                fileExtension={file.extension}
+                actionIcon={
+                  showSpinner ? (
+                    <Spinner size={SpinnerSize.medium} aria-live={'polite'} role={'status'} />
+                  ) : true &&
+                    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ !file.previewUrl ? (
```

Comment:
same thing here, i think using the actual enum type is more reliable. i don't think the assumption `if preview url exist, then apply teams file logic` is solid...

Reply:
In UI Component it has lost the information 'teamsFile' as it will convert 'file' to 'fileSharing' in the binding layer

Reply:
right that's what i meant, i thought we were going to introduce "teamsFile" in addition to "fileSharing"?

Reply:
i don't think the idea of being agnostic makes sense here. At the end of the day, there are two types of files and two set of logics for these types. How does it make sense to differentiate them solely based on the content of preview url property? 

Moreover, there are teams file specific logic on the UI layer as well, I do think a new type is justified, which is something i thought we have agreed yesterday and you even mentioned ARB review for this specifically?

Reply:
Also if we use the previewUrl to differentiate different file types, it might restrict how we can extend the attachment feature. For example, Dominik and I had discussed about how to incorporate the concept of preview Type to attachment files. So that we can have a preview card rendered in addition to attachment files such as this: 
![image](https://github.com/Azure/communication-ui-library/assets/109105353/f229378d-c0ba-4ddd-8f02-a2fcab8d61ea). 

If we were support this in the future, then Teams file attachment logic would be applied to ACS file attachments since both of them would have `previewUrl`

Reply:
For now in the binding layer if attachment type is FileSharing will map previewUrl (sharePoint url) to url, and will still have previewUrl that will be used to hide 'Download' Icon.
Future refactoring task to extract into different FileMetadata arch type will happen in ADO 3242058